server/server.go

```diff
@@ -525,7 +526,7 @@ func (a *ArgoCDServer) newHTTPServer(ctx context.Context, port int, grpcWebHandl
 	httpS := http.Server{
 		Addr: endpoint,
 		Handler: &handlerSwitcher{
-			handler: &bug21955Workaround{handler: mux},
```

Comment:
This works! Just want to suggest a slightly different approach to avoid coupling code related to golang bug workaround:

* Create handler which adds root prefix to any handler:

```golang
func withRootPath(handler http.Handler, root string) http.Handler {
	// get rid of slashes
	root = strings.TrimRight(strings.TrimLeft(root, "/"), "/")

	mux := http.NewServeMux()
	mux.Handle("/"+root+"/", http.StripPrefix("/"+root, handler))
	return mux
}
```

* use it to add root path to argocd server [here](https://github.com/argoproj/argo-cd/blob/f1e0fb936120066db78ec48c20de361f4e20b08d/server/server.go#L247):
```golang
	if rootPath != "" {
		httpS.Handler = withRootPath(httpS.Handler, rootPath)
		if httpsS != nil {
			httpsS.Handler = withRootPath(httpsS.Handler, rootPath)
		}
	}
```